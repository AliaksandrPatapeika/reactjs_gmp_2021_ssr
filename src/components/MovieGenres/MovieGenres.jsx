import './MovieGenres.less';

import PropTypes from 'prop-types';
import React from 'react';

const MovieGenres = ({genres}) => <div className="movieGenres">{genres}</div>;

MovieGenres.propTypes = {
  genres: PropTypes.string.isRequired
};

export default MovieGenres;
