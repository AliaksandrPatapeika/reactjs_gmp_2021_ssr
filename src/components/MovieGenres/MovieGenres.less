.movieGenres {
	font-size: 14px;
}
