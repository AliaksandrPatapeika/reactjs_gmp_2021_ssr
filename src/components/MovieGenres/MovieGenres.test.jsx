import {head} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import MockMovies from '../../tests/mocks/mockData';
import MovieGenres from './MovieGenres';

describe('MovieGenres', () => {
  test('renders correctly', () => {
    const mockMovie = head(MockMovies);
    const movieGenres = renderer
      .create(<MovieGenres genres={mockMovie.genres.join(', ')} />)
      .toJSON();

    expect(movieGenres).toMatchSnapshot();
  });
});
