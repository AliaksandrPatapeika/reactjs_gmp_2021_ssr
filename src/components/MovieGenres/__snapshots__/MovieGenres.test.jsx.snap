// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGenres renders correctly 1`] = `
<div
  className="movieGenres"
>
  Action & Adventure
</div>
`;
